### PR TITLE
Research on different methods for label grouping

### PR DESCRIPTION
**Pull request by SzymonKozl**
Results on method based on layout templates not promising at this moment

### CODE REVIEW

**Comment by SzymonKozl** [comment]
closing this for now as the results seems to be useless

**Comment by SzymonKozl** [comment]
closing this as results seems useless

**Comment by GustawB** [comment]
@SzymonKozl weren't we supposed to merge this at some point?

File: research/preprocessing_szymon/data/18789327023.csv
Comment (GustawB) [inline research/preprocessing_szymon/data/18789327023.csv:1]: Please remove this file.

File: research/preprocessing_szymon/grouping.ipynb [hunk 1]
```diff
@@ -0,0 +1,382 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Grouping by layout templates\n",
+    "##### The purpouse of this notebook was to test different approach to grouping labels\n",
```
Comment (GustawB) [inline research/preprocessing_szymon/grouping.ipynb:8]:
```suggestion
    "##### The purpose of this notebook was to test different approaches to the labels grouping problem\n",
```

File: research/preprocessing_szymon/grouping.ipynb [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    "from vectordb import InMemoryExactNNVectorDB\n",
+    "\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Algorithm hyperparameters"
```
Comment (GustawB) [inline research/preprocessing_szymon/grouping.ipynb:39]:
```suggestion
    "##### Algorithm's hyperparameters"
```

File: research/preprocessing_szymon/grouping.ipynb [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+   "metadata": {},
+   "source": [
+    "## Experiment results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The hyperparameters above were choosen as result of empirical experiments. The results show that this method currently cannot provide robust label grouping in case of sparse data. Research on larger datasets to be conducted."
```
Comment (GustawB) [inline research/preprocessing_szymon/grouping.ipynb:352]: I know that I wasn't supposed to be requesting everything, BUT for the sake of completeness I would elaborate why those results are bad.  At least at first glance, these results doesn't look that bad (at least in the context of out recent results on PoC).

**Review by GustawB — CHANGES_REQUESTED** [review]

File: research/preprocessing_szymon/grouping.ipynb [hunk 4]
```diff
@@ -0,0 +1,392 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Grouping by layout templates\n",
+    "##### The purpouse of this notebook was to test different approach to grouping labels\n",
+    "Firtly we compute distances between neighbour labels. Then we aggregate them and count duplicated vectors. To handle vector storing and querying I used vectorDB. Finally, I created groups by connecting labels if distance between them is common."
```
Comment (GustawB) [inline research/preprocessing_szymon/grouping.ipynb:9]:
```suggestion
    "Firstly we compute distances between neighbour labels. Then we aggregate them and count duplicated vectors. To handle vector storing and querying I used vectorDB. Finally, I created groups by connecting labels if distance between them is common."
```

**Review by GustawB — CHANGES_REQUESTED** [review]

**Review by GustawB — APPROVED** [review]
LGTM

**Pull request merged**